webapp/channels/src/components/new_search/search_box_input.tsx

```diff
@@ -0,0 +1,136 @@
+import {useDispatch} from 'react-redux';
+import styled from 'styled-components';
+
+import {
+    updateSearchTerms,
+    updateSearchType,
+} from 'actions/views/rhs';
+
+import QuickInput from 'components/quick_input';
+
+const SearchInputContainer = styled.div`
+    position: relative;
+    display: flex;
+    align-items: center;
+    i {
+        color: var(--center-channel-color-56);
+        display: flex;
+        align-items: center;
+        &.icon-close {
+            postion: absolute;
+            right: 10px;
+        }
+        &.icon-magnify {
+            position: absolute;
+            left: 20px;
+            top: 21px;
+            font-size: 24px;
+        }
+    }
+    .input-wrapper {
+        flex-grow: 1;
+    }
+    && input {
+        padding: 20px 100px 20px 56px;
+        height: auto;
+        border-radius: 0;
+        border: none;
+        border-bottom: var(--border-default);
+        font-size: 20px;
+        line-height: 28px;
+        font-family: Metropolis, sans-serif;
+        :focus {
+            border: none;
+            border-bottom: var(--border-default);
+            box-shadow: none;
+        }
+        ::placeholder {
+            color: rgba(var(--center-channel-color-rgb), 0.75);
+        }
+    }
+`;
+
+const ClearButton = styled.button`
+    display: flex;
+    position: absolute;
+    right: 12px;
+    background: none;
+    color: rgba(var(--center-channel-color-rgb), 0.75);
+    &:hover{
+        color: rgba(var(--center-channel-color-rgb), 0.88);
+        background: rgba(var(--center-channel-color-rgb), 0.08);
+    }
+`;
+
+type Props = {
+    searchTerms: string;
+    searchType: string;
+    setSearchTerms: (searchTerms: string) => void;
+    onKeyDown: (e: React.KeyboardEvent<Element>) => void;
+    focus: (newPosition: number) => void;
+}
+
+const SearchInput = ({searchTerms, searchType, setSearchTerms, onKeyDown, focus}: Props, inputRef: React.Ref<HTMLInputElement>) => {
+    const intl = useIntl();
+    let searchPlaceholder = intl.formatMessage({id: 'search_bar.search', defaultMessage: 'Search'});
+
+    if (searchType === 'messages' || searchType === '') {
+        searchPlaceholder = intl.formatMessage({id: 'search_bar.search_messages', defaultMessage: 'Search messages'});
+    }
+    if (searchType === 'files') {
+        searchPlaceholder = intl.formatMessage({id: 'search_bar.search_files', defaultMessage: 'Search files'});
+    }
+
+    const dispatch = useDispatch();
+
+    const inputChangeCallback = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+        setSearchTerms(e.target.value);
+    }, [setSearchTerms]);
+
+    const clearSearch = useCallback(() => {
+        setSearchTerms('');
+        dispatch(updateSearchTerms(''));
+        dispatch(updateSearchType('messages'));
+        focus(0);
+    }, [focus, setSearchTerms]);
+
+    return (
+        <SearchInputContainer>
+            <i className='icon icon-magnify'/>
+            <QuickInput
```

Comment:
Does this actually need to be a `QuickInput` or could it just be a regular `input`? The only real reason we need it any more is because someone added a clear button to it, but this component implements its own clear button already

Reply:
I kept this input to be sure that I was not breaking anything in terms of backward compatibility or accesibility. We can remove it later, but I don't want to accidentally break something important. If you feel like this should be done in this PR, I'm ok on doing that.

Reply:
Yeah, we can change that later on